core-bundle/src/Resources/contao/modules/Module.php

```diff
@@ -342,11 +343,29 @@ protected function renderNavigation($pid, $level=1, $host=null, $language=null)
 							continue 2;
 						}
 
-						$href = $objNext->getFrontendUrl();
+						try
+						{
+							$href = $objNext->getFrontendUrl();
+						}
+						catch (ExceptionInterface $exception)
+						{
+							System::log('Unable to generate the URL for page ID ' . $objNext->id . ': ' . $exception->getMessage(), __METHOD__, TL_ERROR);
```

Comment:
Does it really make sense to log this? Because if it is the expected behavior (e.g. "require an item"), it is neither an error nor could the user do anything about it.

Reply:
I think she can, the page should not be in the navigation in the first place.

Reply:
You are right. 👍🏻